cases/serializers.py

```
@@ -653,9 +666,27 @@ class Meta:
+
+        The file is NOT persisted to the ``uploaded_file`` FileField; ``url`` is
+        the single source of truth for a source's links.
+        """
+        from cases.services.source_files import store_file_as_link
+
+        uploaded_file = validated_data.pop("uploaded_file", None)
+        upload_role = validated_data.pop("upload_role", SourceLinkRole.RAW.value)
+
+        if uploaded_file is not None:
+            link = store_file_as_link(uploaded_file, role=upload_role)
+            validated_data["url"] = list(validated_data.get("url") or []) + [link]
```

Ruff (0.15.15)

[warning] Consider `[*list(validated_data.get("url") or []), link]` instead of concatenation

Replace with `[*list(validated_data.get("url") or []), link]`

(RUF005)